js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+				$scope.users = response.data.data.items;
+
+				// Handle pagination
+				if (!$scope.useCustomFilter && (selectedUsersCount === 0)) {
+					// If there are more results than MAX_COUNT, launch a timer to get the number of results
+					if ($scope.users.length > MAX_COUNT) {
+						countTimeout(input);
+						$scope.users = _.first($scope.users, MAX_COUNT);
+						$scope.users.push({ overflow: MAX_COUNT + "/..." });
+					}
+				}
+				else {
+					// Used for UserPickerMultiple only
+					if (selectedUsersCount > 0) {
+						// Remove duplicates between results and selected users
+						_.each($scope.selected.users, function (selectedUser) {
+							$scope.users = _.reject($scope.users, function (user) {
+								return (user.id === selectedUser.id);
+							});
+						});
+						// Number of results matching the query
+						$scope.count = $scope.users.length;
+					}
+
+					// Used when a custom filtering function is given
+					if ($scope.useCustomFilter) {
+						// Get new set of users
+						$scope.users = $scope.customFilter({ users: $scope.users });
+						$scope.count = $scope.users.length;
+					}
+
+					if ($scope.count > MAX_COUNT) {
+						$scope.users = _.first($scope.users, MAX_COUNT);
+						$scope.users.push({ overflow: MAX_COUNT + "/" + $scope.count });
+					}
+				}
+
+				if ($scope.showFormerEmployees) {
+					checkFormerEmployees();
+				}
+
+				if (hasHomonyms()) {
+					getHomonymsProperties();
+				}
+			});
+		};
+
+		var checkFormerEmployees = function () {
+			_.each($scope.users, function (user) {
+				if (moment(user.dtContractEnd).isBefore(moment())) {
+					user.isFormerEmployee = true;
+				}
+			});
+		};
+
+		var hasHomonyms = function () {
+			var hasHomonyms = false;
+
+			_.each($scope.users, function (user) {
+								// Search homonyms in set of results
+								_.each($scope.users, function (otherUser) {
+									if ((user.firstName === otherUser.firstName) && (user.lastName === otherUser.lastName) && (user.id !== otherUser.id)) {
+										user.hasHomonyms = true;
+										otherUser.hasHomonyms = true;
+										if (!hasHomonyms) {
+											hasHomonyms = true;
+										}
+									}
+								});
+
+								// Search homonyms in selected users (used by UserPickerMultiple only)
+								_.each($scope.selected.users, function (selectedUser) {
+									if ((user.firstName === selectedUser.firstName) && (user.lastName === selectedUser.lastName)) {
+										user.hasHomonyms = true;
+										selectedUser.hasHomonyms = true;
+										if (!hasHomonyms) {
+											hasHomonyms = true;
+										}
+									}
+								});
+							});
+			return hasHomonyms;
+		};
+
+		var getHomonymsProperties = function () {
+			var homonyms = _.where($scope.users, { hasHomonyms: true });
+			var homonymsArray = [];
+
+			loadPropertiesFromUrls(homonyms).then(function (response) {
+								// Add homonyms properties for each user
+								_.each(homonyms, function (user, item) {
```

Comment:
this indentation if [off the hizzle](http://www.urbandictionary.com/define.php?term=hizzle&defid=1138738)